### PR TITLE
[V2 experimentation]profiler: remove WithAPIKey and WithAgentlessUpload

### PR DESCRIPTION
**Pull request by JianyiGao**
### What does this PR do?
Remove `WithAPIKey` option 
https://github.com/DataDog/dd-trace-go/issues/866

### Motivation
TLDR: 
Agentless uploading is originally supported if customer set `DD_API_KEY` env variable, or use `WithAPIKey` option. [`WithAgentlessUpload` was introduced for debugging purpose](https://github.com/DataDog/dd-trace-go/pull/906/) when agentless mode is disabled for `WithAPIKey` option.

Due to business reasons we want to deprecate agentless support for public uses, only allow internal usage by explicitly set `DD_PROFILING_AGENTLESS` and `DD_API_KEY` env variable. 

This PR is to remove `WithAPIKey` and `WithAgentlessUpload` option.

More context: [Agentless vs Agent Uploading](https://docs.google.com/document/d/172cmlRA_GTSu2OjC_FL5PzO6EjCpRH5ddU1nP08-yF0/edit)

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Review by felixge — CHANGES_REQUESTED** [review]
This is a breaking API change. Please ping me in slack to provide more context on this. Thanks!

cc @nsrip-dd

**Review by darccio — APPROVED** [review]

File: profiler/options.go [hunk 1]
```diff
@@ -319,19 +319,6 @@ func WithAgentAddr(hostport string) Option {
...
-// currently don't have an agent running on the default localhost:8126 hostport
-// you need to set it up, or use WithAgentAddr to specify the hostport location
-// of the agent. See WithAgentlessUpload for more information.
-func WithAPIKey(key string) Option {
-	return func(cfg *config) {
-		cfg.apiKey = key
-	}
-}
-
 // WithAgentlessUpload is currently for internal usage only and not officially
```
Comment (felixge) [inline profiler/options.go:322]: While we're at it, I think we should probably remove `WithAgentlessUpload` and replace it with an environment variable as well?

**Review by felixge — APPROVED** [review]

File: profiler/options.go [hunk 2]
```diff
@@ -246,10 +246,10 @@ func defaultConfig() (*config, error) {
 		WithUploadTimeout(d)(&c)
 	}
 	if v := os.Getenv("DD_API_KEY"); v != "" {
-		WithAPIKey(v)(&c)
+		c.apiKey = v
 	}
 	if internal.BoolEnv("DD_PROFILING_AGENTLESS", false) {
-		WithAgentlessUpload()(&c)
+		c.agentless = true
 	}
```
Comment (felixge) [inline profiler/options.go:253]: NIT: Isn't this the same as `c.agentless = internal.BoolEnv("DD_PROFILING_AGENTLESS", false)`?

**Review by felixge — APPROVED** [review]
LGTM
